rocket-okapi/src/request/from_query_multi_param_impls.rs

```diff
@@ -0,0 +1,153 @@
+use super::OpenApiFromQuery;
+use crate::gen::OpenApiGenerator;
+use okapi::openapi3::*;
+use std::result::Result as StdResult;
+use schemars::JsonSchema;
+use schemars::schema::{Schema, ObjectValidation, SchemaObject};
+use std::collections::BTreeMap;
+
+type Result = crate::Result<Vec<Parameter>>;
+
+/// Given an object that implements the `JsonSchema` generate all the `Parameter`
+/// that are used to create documentation.
+/// Use when manualy implementing a
+/// [Query Guard](https://docs.rs/rocket/latest/rocket/request/trait.FromQuery.html).
+/// Example:
+/// ```
+/// use rocket::request::{Query, FromQuery};
+/// use serde::{Serialize, Deserialize};
+/// use schemars::JsonSchema;
+/// use rocket_okapi::{
+///     gen::OpenApiGenerator,
+///     request::OpenApiFromQuery,
+///     request::get_nested_query_parameters
+/// };
+///
+/// #[derive(Serialize, Deserialize, Clone, Debug, Default, JsonSchema)]
+/// pub struct ApiPagination{
+///     page: Option<u32>,
+///     per_page: Option<u32>,
+/// }
+///
+/// impl<'q> FromQuery<'q> for ApiPagination {
+///     type Error = String;// Some kind of error
+///
+///     fn from_query(_query: Query<'q>) -> Result<Self, Self::Error> {
+///         Ok(ApiPagination::default())
+///     }
+/// }
+///
+/// impl<'r> OpenApiFromQuery<'r> for ApiPagination {
+///     fn query_multi_parameter(gen: &mut OpenApiGenerator, name: String, required: bool)
+///     -> rocket_okapi::Result<Vec<okapi::openapi3::Parameter>> {
+///         Ok(get_nested_query_parameters::<ApiPagination>(gen, name, required))
+///     }
+/// }
+/// ```
+pub fn get_nested_query_parameters<'r, T>(gen: &mut OpenApiGenerator, name: String, required: bool)
+-> Vec<Parameter> where
+    T: JsonSchema{
+    let schema = gen.json_schema::<T>();
+    let mut referenced_schema: Option<Schema> = None;
+    if let Some(reference) = &schema.reference{
+        referenced_schema = gen.json_definition(reference.clone());
+    }
+    if let Some(schema) = referenced_schema{
+        // Get a list of properties from the structure.
+        let mut properties: BTreeMap<String, Schema> = BTreeMap::new();
+        match schema {
+            Schema::Object(schema) => {
+                if let Some(object) = schema.object{
+                    let object: &'r mut ObjectValidation = Box::leak(object);
+                    properties = object.properties.clone();
```

Comment:
I don't understand why we're leaking the box here? Also, since the schema is owned rather than borrowed, there's no need to clone the properties - we can just take them:
```suggestion
                    properties = object.properties;
```

Reply:
I'm not very familiar with `Box` yet. But this code was no longer needed after changing to `json_schema_no_ref`. So that resolves this too.